rubin_sim/maf/metrics/color_slope_metrics.py

```diff
@@ -0,0 +1,188 @@
+        to still count as a color (hours). Default 1 hour.
+    slope_length : `float`
+        The length of time to demand observations in the
+        same filter be greater than (hours). Default 3 hours.
+    """
+
+    def __init__(
+        self, color_length=1.0, slope_length=3.0, filter_col="filter", mjd_col="observationStartMJD"
+    ):
+        self.color_length = color_length / 24.0
+        self.slope_length = slope_length / 24.0
+
+        self.filter_col = filter_col
+        self.mjd_col = mjd_col
+
+    def __call__(self, data_slice):
+        has_color = False
+        has_slope = False
+
+        if np.size(data_slice) < 3:
+            return 0
+        filters = data_slice[self.filter_col]
+
+        u_filters = np.unique(filters)
+
+        for filtername in u_filters:
+            in_filt = np.where(data_slice[self.filter_col] == filtername)[0]
+            time_gap = (
+                data_slice[self.mjd_col][in_filt].max() - data_slice[self.mjd_col][in_filt][np.newaxis].min()
+            )
+            if time_gap >= self.slope_length:
+                has_slope = True
+                break
+        for filtername1 in u_filters:
+            for filtername2 in u_filters:
+                if filtername1 != filtername2:
+                    in_filt1 = np.where(filters == filtername1)[0]
+                    in_filt2 = np.where(filters == filtername2)[0]
+                    time_gaps = (
+                        data_slice[self.mjd_col][in_filt1] - data_slice[self.mjd_col][in_filt2][np.newaxis].T
+                    )
+                    time_gaps = time_gaps[np.where(time_gaps > 0)]
+                    if time_gaps.size > 0:
+                        if np.min(time_gaps[np.where(time_gaps > 0)]) <= self.color_length:
+                            has_color = True
+                            break
+        if has_color & has_slope:
+            return 1
+        else:
+            return 0
+
+
+class ColorSlopeMetric(BaseMetric):
+    """How many times do we get a color and slope in a night
+    A proxie metric for seeing how many times
+    there would be the possibility of identifying and
+    classifying a transient.
+
+    Parameters
+    ----------
+    mag : `dict`
+        Dictionary with filternames as keys and minimum depth m5
+        magnitudes as values. If None, defaults to mag 20 in ugrizy.
+    color_length : `float`
+        The maximum length of time different filters be observed
+        to still count as a color (hours). Default 1 hour.
+    slope_length : `float`
+        The length of time to demand observations in the
+        same filter be greater than (hours). Default 3 hours."""
+
+    def __init__(
+        self,
+        mag=None,
+        night_col="night",
+        filter_col="filter",
+        m5_col="fiveSigmaDepth",
+        color_length=1.0,
+        slope_length=3.0,
+        time_col="observationStartMJD",
+        units="#",
+        metric_name="ColorSlope",
```

Comment:
I'm curious because I see this pattern a lot -- but why specify the metric_name here when it's the same as what will be constructed by default?

Reply:
I bet I was probably planning on having one metric with a kwarg rather then two, so I'd be setting the metric_name to run it different ways.